prowler/providers/aws/aws_provider_new.py

```diff
@@ -0,0 +1,622 @@
+        )
+
+        # Set ignore unused services
+        self.ignore_unused_services = ignore_unused_services
+
+        # Start populating AWS identity object
+        self.identity.profile = input_profile
+        self.identity.audited_regions = input_regions
+
+        # We need to create an original sessions using regular auth path (creds, profile, etc)
+        logger.info("Generating original session ...")
+        self.session.session = self.setup_session(input_mfa)
+
+        # After the session is created, validate it
+        logger.info("Validating credentials ...")
+        caller_identity = validate_aws_credentials(
+            self.session.session, input_regions, sts_endpoint_region
+        )
+
+        logger.info("Credentials validated")
+        logger.info(f"Original caller identity UserId: {caller_identity['UserId']}")
+        logger.info(f"Original caller identity ARN: {caller_identity['Arn']}")
+        # Set values of AWS identity object
+        self.identity.account = caller_identity["Account"]
+        self.identity.identity_arn = caller_identity["Arn"]
+        self.identity.user_id = caller_identity["UserId"]
+        self.identity.partition = parse_iam_credentials_arn(
+            caller_identity["Arn"]
+        ).partition
+        self.identity.account_arn = (
+            f"arn:{self.identity.partition}:iam::{self.identity.account}:root"
+        )
+
+        # save original session
+        self.session.original_session = self.session.session
+        # time for checking role assumption
+        if input_role:
+            # session will be the assumed one
+            self.session.session = self.setup_assumed_session(
+                input_role,
+                input_external_id,
+                input_mfa,
+                input_session_duration,
+                sts_endpoint_region,
+            )
+            logger.info("Audit session is the new session created assuming role")
+        # check if organizations info is gonna be retrieved
+        if organizations_role_arn:
+            logger.info(
+                f"Getting organizations metadata for account {organizations_role_arn}"
+            )
+            # session will be the assumed one with organizations permissions
+            self.session.session = self.setup_assumed_session(
+                organizations_role_arn,
+                input_external_id,
+                input_mfa,
+                input_session_duration,
+                sts_endpoint_region,
+            )
+            self.organizations_metadata = get_organizations_metadata(
+                self.identity.account, self.assumed_role.assumed_role_credentials
+            )
+            logger.info("Organizations metadata retrieved")
+        if self.session.session.region_name:
+            self.identity.profile_region = self.session.session.region_name
+        else:
+            self.identity.profile_region = "us-east-1"
+
+        if not arguments.get("only_logs"):
```

Comment:
I think this should be out of the init. One of the main ideas to make a generic provider is to only call the provider `print_credentials` in one place without the need to include that in the `init`.

Reply:
Right now the class CloudProvider does nothing
```
class CloudProvider(ABC):
    @abstractmethod
    def setup_session(self):
        pass

    @abstractmethod
    def print_credentials(self):
        pass

    def validate_arguments(self):
        pass
```

This needs to be the one in charge to execute the logic to instantiate the provider.

Reply:
And we should rename `CloudProvider` to `Provider`,

Reply:
name changed, can we do the generic calls in other PR?